x-pack/filebeat/input/cometd/input_test.go

```diff
@@ -0,0 +1,279 @@
+	}
+	assert.NotEqual(t, event, makeEvent("DEMOID", "DEMOBODY"))
+}
+
+func TestNewInput_Run(t *testing.T) {
+	eventsCh := make(chan beat.Event)
+
+	outlet := &mockedOutleter{
+		onEventHandler: func(event beat.Event) bool {
+			eventsCh <- event
+			return true
+		},
+	}
+	connector := &mockedConnector{
+		outlet: outlet,
+	}
+	var inputContext finput.Context
+
+	var msg bay.TriggerEvent
+	msg.Data.Event.ReplayID = 1234
+	msg.Data.Payload = []byte(`{"CountryIso": "IN"}`)
+	msg.Channel = channelName
+
+	config := map[string]interface{}{
+		"channel_name":              channelName,
+		"auth.oauth2.client.id":     "client.id",
+		"auth.oauth2.client.secret": "client.secret",
+		"auth.oauth2.user":          "user",
+		"auth.oauth2.password":      "password",
+	}
+
+	r := http.HandlerFunc(oauth2Handler)
+	server := httptest.NewServer(r)
+	serverURL = server.URL
+	config["auth.oauth2.token_url"] = server.URL + "/token"
+
+	cfg := common.MustNewConfigFrom(config)
+
+	input, err := NewInput(cfg, connector, inputContext)
+	require.NoError(t, err)
+	require.NotNil(t, input)
+
+	input.Run()
+	for _, event := range []beat.Event{<-eventsCh} {
+		require.NoError(t, err)
+		assertEventMatches(t, msg, event)
+	}
+	server.Close()
+}
+
+func TestNewInput_Run_Wait(t *testing.T) {
+	eventsCh := make(chan beat.Event)
+
+	outlet := &mockedOutleter{
+		onEventHandler: func(event beat.Event) bool {
+			eventsCh <- event
+			return true
+		},
+	}
+	connector := &mockedConnector{
+		outlet: outlet,
+	}
+	var inputContext finput.Context
+
+	var msg bay.TriggerEvent
+	msg.Data.Event.ReplayID = 1234
+	msg.Data.Payload = []byte(`{"CountryIso": "IN"}`)
+	msg.Channel = channelName
+
+	config := map[string]interface{}{
+		"channel_name":              channelName,
+		"auth.oauth2.client.id":     "client.id",
+		"auth.oauth2.client.secret": "client.secret",
+		"auth.oauth2.user":          "user",
+		"auth.oauth2.password":      "password",
+	}
+
+	r := http.HandlerFunc(oauth2Handler)
+	server := httptest.NewServer(r)
+	serverURL = server.URL
+	config["auth.oauth2.token_url"] = server.URL + "/token"
+
+	cfg := common.MustNewConfigFrom(config)
+
+	input, err := NewInput(cfg, connector, inputContext)
+	require.NoError(t, err)
+	require.NotNil(t, input)
+
+	input.Run()
+
+	go func() {
+		time.Sleep(100 * time.Millisecond) // let input.Stop() be executed.
+		for range eventsCh {
+		}
+	}()
+
+	input.Wait()
+}
+
+func TestStop(t *testing.T) {
+	conf := defaultConfig()
+	logger := logp.NewLogger("test")
+	authParams := bay.AuthenticationParameters{}
+	inputCtx, cancelInputCtx := context.WithCancel(context.Background())
+	workerCtx, workerCancel := context.WithCancel(inputCtx)
+	defer cancelInputCtx()
+
+	input := &cometdInput{
+		config:       conf,
+		log:          logger,
+		inputCtx:     inputCtx,
+		workerCtx:    workerCtx,
+		workerCancel: workerCancel,
+		ackedCount:   atomic.NewUint32(0),
+		authParams:   authParams,
+	}
+	input.out = make(chan bay.TriggerEvent)
+
+	input.Stop()
```

Comment:
There are no assertions or calls to `t.Fail` in this test. I don't think it could fail. You would need a way to check that the input is actually stopped (looking at the context perhaps?)

Reply:
Same for `TestWait` below.